storage/postgres/security.go

```diff
@@ -92,6 +92,7 @@ type keyFetcher struct {
 func (s *keyFetcher) GetEncryptionKey(ctx context.Context) ([]byte, error) {
```

Comment:
Can we pass here the pgDB object, so we reuse the already opened transaction connection? This way we won't even open another connection?

Reply:
The last screenshot shows just that. Instead of changing the interface, just for the tests I did:
```go
        var objectList types.ObjectList
	err := itr.InTransaction(ctx, func(ctx context.Context, txStorage Repository) error {
		var err error
		objectList, err = txStorage.List(ctx, objectType, criteria...)
		if err != nil {
			return err
		}
		encrypter := &security.TwoLayerEncrypter{Fetcher: txStorage.Security().Fetcher()}
		for i := 0; i < objectList.Len(); i++ {
			obj := objectList.ItemAt(i)
			if err = transformCredentials(ctx, obj, encrypter.Decrypt); err != nil {
				return err
			}
		}
		return nil
	})
	if err != nil {
		return nil, err
	}

	return objectList, nil
```
Because of the underlying implementation of storage.Security(), the pgDB object is actually `sqlx.Tx` object. https://github.com/Peripli/service-manager/blob/master/storage/postgres/storage.go#L293 and https://github.com/Peripli/service-manager/blob/master/storage/postgres/storage.go#L60

Reply:
Yes, that is want I meant by pgDB object, the `sqlx.Tx`

Reply:
How many parallel requests could you do after the resource leak fix?